Index: sdl-gpu/demos/CMakeLists.txt
===================================================================
--- sdl-gpu.orig/demos/CMakeLists.txt
+++ sdl-gpu/demos/CMakeLists.txt
@@ -13,7 +13,7 @@ target_link_libraries (init-demo ${DEMO_
 add_executable(camera-demo camera/main.c)
 target_link_libraries (camera-demo ${DEMO_LIBS})
 
-add_executable(3d-demo 3d/main.c ../src/externals/glew/glew.c)
+add_executable(3d-demo 3d/main.c)
 target_link_libraries (3d-demo ${DEMO_LIBS})
 
 add_executable(tutorial-space-demo tutorial-space/main.c)
Index: sdl-gpu/demos/3d/main.c
===================================================================
--- sdl-gpu.orig/demos/3d/main.c
+++ sdl-gpu/demos/3d/main.c
@@ -1,7 +1,7 @@
 #include "SDL.h"
 #include "SDL_gpu.h"
 #include "SDL_gpu_OpenGL_1.h"
-#include "glew.h"
+#include <GL/glew.h>
 #include <stdlib.h>
 
 
Index: sdl-gpu/include/SDL_gpu_OpenGL_1.h
===================================================================
--- sdl-gpu.orig/include/SDL_gpu_OpenGL_1.h
+++ sdl-gpu/include/SDL_gpu_OpenGL_1.h
@@ -11,7 +11,7 @@
     #define _WINGDI_H
     #endif
 
-    #include "glew.h"
+    #include <GL/glew.h>
 	
 	#if defined(GL_EXT_bgr) && !defined(GL_BGR)
 		#define GL_BGR GL_BGR_EXT
Index: sdl-gpu/include/SDL_gpu_OpenGL_1_BASE.h
===================================================================
--- sdl-gpu.orig/include/SDL_gpu_OpenGL_1_BASE.h
+++ sdl-gpu/include/SDL_gpu_OpenGL_1_BASE.h
@@ -11,7 +11,7 @@
     #define _WINGDI_H
     #endif
 
-    #include "glew.h"
+    #include <GL/glew.h>
 	
 	#if defined(GL_EXT_bgr) && !defined(GL_BGR)
 		#define GL_BGR GL_BGR_EXT
Index: sdl-gpu/include/SDL_gpu_OpenGL_2.h
===================================================================
--- sdl-gpu.orig/include/SDL_gpu_OpenGL_2.h
+++ sdl-gpu/include/SDL_gpu_OpenGL_2.h
@@ -11,7 +11,7 @@
     #define _WINGDI_H
     #endif
     
-    #include "glew.h"
+    #include <GL/glew.h>
 	
 	#if defined(GL_EXT_bgr) && !defined(GL_BGR)
 		#define GL_BGR GL_BGR_EXT
Index: sdl-gpu/include/SDL_gpu_OpenGL_3.h
===================================================================
--- sdl-gpu.orig/include/SDL_gpu_OpenGL_3.h
+++ sdl-gpu/include/SDL_gpu_OpenGL_3.h
@@ -11,7 +11,7 @@
     #define _WINGDI_H
     #endif
     
-    #include "glew.h"
+    #include <GL/glew.h>
 	
 	#if defined(GL_EXT_bgr) && !defined(GL_BGR)
 		#define GL_BGR GL_BGR_EXT
Index: sdl-gpu/include/SDL_gpu_OpenGL_4.h
===================================================================
--- sdl-gpu.orig/include/SDL_gpu_OpenGL_4.h
+++ sdl-gpu/include/SDL_gpu_OpenGL_4.h
@@ -11,7 +11,7 @@
     #define _WINGDI_H
     #endif
     
-    #include "glew.h"
+    #include <GL/glew.h>
 	
 	#if defined(GL_EXT_bgr) && !defined(GL_BGR)
 		#define GL_BGR GL_BGR_EXT
