Index: sdl-gpu/src/CMakeLists.txt
===================================================================
--- sdl-gpu.orig/src/CMakeLists.txt
+++ sdl-gpu/src/CMakeLists.txt
@@ -162,7 +162,8 @@ if(BUILD_SHARED)
 			set_target_properties(SDL_gpu_shared PROPERTIES
 				OUTPUT_NAME ${SHARED_OUTPUT}
 				CLEAN_DIRECT_OUTPUT 1
-#				VERSION ${SDL_GPU_VERSION}
+				VERSION ${SDL_GPU_VERSION}
+				SOVERSION ${VERSION_MAJOR}
 				ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${OUTPUT_DIR}/lib"
 				LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${OUTPUT_DIR}/lib"
 				RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${OUTPUT_DIR}/bin"
